crates/rattler_virtual_packages/src/lib.rs

```diff
@@ -48,35 +47,81 @@ use libc::DetectLibCError;
 use linux::ParseLinuxVersionError;
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
 
+/// Configure the overrides used in in this crate.
+#[derive(Clone, Debug, PartialEq)]
+pub enum Override {
+    /// Use the default override env var name
+    DefaultEnvVar,
+    /// Use custom env var name
+    EnvVar(String),
+    /// Use a custom override directly
+    String(String),
+    /// Disable overrides
+    None,
+}
+
+impl Default for Override {
+    fn default() -> Self {
+        Self::DefaultEnvVar
+    }
+}
+
 /// Traits for overridable virtual packages
 /// Use as `Cuda::from_default_env_var.unwrap_or(Cuda::current().into()).unwrap()`
 pub trait EnvOverride: Sized {
     /// Parse `env_var_value`
-    fn from_env_var_name_with_var(
-        env_var_name: &str,
-        env_var_value: &str,
-    ) -> Result<Self, ParseVersionError>;
+    fn parse_version(value: &str) -> Result<Self, ParseVersionError>;
+
+    /// Helper to convert the output of `parse_version` and handling empty strings.
+    fn parse_version_(value: &str) -> Result<Option<Self>, DetectVirtualPackageError> {
+        if value.is_empty() {
+            Ok(None)
+        } else {
+            Ok(Some(Self::parse_version(value)?))
+        }
+    }
```

Comment:
Maybe rename to `parse_version_opt` (`opt` for optional)? The underscore tells me very little.